From da9b22a416b85b8e4e3166841032a7c8ab07fb0c Mon Sep 17 00:00:00 2001
From: Stuart Morgan <smorgan@mythtv.org>
Date: Mon, 9 Sep 2013 19:23:56 +0100
Subject: [PATCH] Group albums by album name, not album_id. An album to
 mythmusic uses the artist > album hierachy, but for albums
 which have multiple artists this means we have multiple
 entries in the album table. It's a failing in the way we
 arrange the data internally, but it's better to have just
 one entry per album in the upnp output than 10-15 for each
 artist in a compilation.

---
 mythtv/programs/mythbackend/upnpcdsmusic.cpp |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mythtv/programs/mythbackend/upnpcdsmusic.cpp b/mythtv/programs/mythbackend/upnpcdsmusic.cpp
index cd0111f..0d4396a 100644
--- a/mythtv/programs/mythbackend/upnpcdsmusic.cpp
+++ b/mythtv/programs/mythbackend/upnpcdsmusic.cpp
@@ -77,7 +77,7 @@ UPnpCDSRootInfo UPnpCDSMusic::g_RootNodes[] =
           "count( song.album_id ) as children "
             "FROM music_songs song join music_albums a on a.album_id = song.album_id "
             "%1 "
-            "GROUP BY a.album_id "
+            "GROUP BY a.album_name "
             "ORDER BY a.album_name",
         "WHERE song.album_id=:KEY", "album.album_name" },
 
-- 
1.7.10.2

